--- a/ipa-client/ipa-install/ipa-client-install
+++ b/ipa-client/ipa-install/ipa-client-install
@@ -493,10 +493,11 @@
     # Always start certmonger. We can't untrack something if it isn't
     # running
     messagebus = services.knownservices.messagebus
-    try:
-        messagebus.start()
-    except Exception, e:
-        log_service_error(messagebus.service_name, 'start', e)
+    if not messagebus.is_running():
+        try:
+            messagebus.start()
+        except Exception, e:
+            log_service_error(messagebus.service_name, 'start', e)
 
     cmonger = services.knownservices.certmonger
     try:
@@ -1071,10 +1072,11 @@
     principal = 'host/%s@%s' % (hostname, cli_realm)
 
     messagebus = services.knownservices.messagebus
-    try:
-        messagebus.start()
-    except Exception, e:
-        log_service_error(messagebus.service_name, 'start', e)
+    if not messagebus.is_running():
+        try:
+            messagebus.start()
+        except Exception, e:
+            log_service_error(messagebus.service_name, 'start', e)
 
     # Ensure that certmonger has been started at least once to generate the
     # cas files in /var/lib/certmonger/cas.
