--- a/ipaserver/install/httpinstance.py
+++ b/ipaserver/install/httpinstance.py
@@ -121,8 +121,8 @@ class HTTPInstance(service.Service):
                   self.configure_mod_ssl_certs)
         self.step("setting mod_ssl protocol list to TLSv1.0 - TLSv1.2",
                   self.set_mod_ssl_protocol)
-        self.step("configuring mod_ssl log directory",
-                  self.set_mod_ssl_logdir)
+#        self.step("configuring mod_ssl log directory",
+#                  self.set_mod_ssl_logdir)
         self.step("disabling mod_ssl OCSP", self.disable_mod_ssl_ocsp)
         self.step("adding URL rewriting rules", self.__add_include)
         self.step("configuring httpd", self.__configure_http)
@@ -625,7 +625,7 @@ class HTTPInstance(service.Service):
         self.backup_ssl_conf()
         self.configure_mod_ssl_certs()
         self.set_mod_ssl_protocol()
-        self.set_mod_ssl_logdir()
+#        self.set_mod_ssl_logdir()
         self.__add_include()
 
         self.cert = x509.load_certificate_from_file(paths.HTTPD_CERT_FILE)
