--- a/ipaplatform/debian/tasks.py
+++ b/ipaplatform/debian/tasks.py
@@ -57,6 +57,10 @@ class DebianTaskNamespace(RedHatTaskName
         # Debian doesn't use authconfig, nothing to restore
         return True
 
+    def migrate_auth_configuration(self, statestore):
+        # Debian doesn't have authselect
+        return True
+
     @staticmethod
     def parse_ipa_version(version):
         return BaseTaskNamespace.parse_ipa_version(version)
