Description: we have a newer pykerberos than Fedora
--- a/ipalib/rpc.py
+++ b/ipalib/rpc.py
@@ -544,7 +544,7 @@ class KerbTransport(SSLTransport):
         service = "HTTP@" + host.split(':')[0]
 
         try:
-            (rc, vc) = kerberos.authGSSClientInit(service, self.flags)
+            (rc, vc) = kerberos.authGSSClientInit(service, gssflags=self.flags)
         except kerberos.GSSError, e:
             self._handle_exception(e)
 
