--- a/install/share/opendnssec_conf.template
+++ b/install/share/opendnssec_conf.template
@@ -28,8 +28,8 @@
 
 	<Enforcer>
 		<Privileges>
-			<User>ods</User>
-			<Group>ods</Group>
+			<User>$ODS_USER</User>
+			<Group>$ODS_GROUP</Group>
 		</Privileges>
 
 		<Datastore><SQLite>$KASP_DB</SQLite></Datastore>
--- a/ipaserver/install/opendnssecinstance.py
+++ b/ipaserver/install/opendnssecinstance.py
@@ -80,6 +80,8 @@ class OpenDNSSECInstance(service.Service
             'SOFTHSM_LIB': paths.LIBSOFTHSM2_SO,
             'TOKEN_LABEL': dnskeysyncinstance.softhsm_token_label,
             'KASP_DB': paths.OPENDNSSEC_KASP_DB,
+            'ODS_USER': ODS_USER,
+            'ODS_GROUP': ODS_GROUP,
         }
         self.kasp_file_dict = {}
         self.extra_config = [KEYMASTER]
