--- a/install/ui/util/build.sh
+++ b/install/ui/util/build.sh
@@ -31,6 +31,6 @@ if [[ ! $profile ]] ; then
     exit 1
 fi
 
-RHINO="java -Xss${JAVA_STACK_SIZE:-512k} -classpath /usr/share/java/rhino.jar  org.mozilla.javascript.tools.shell.Main"
+RHINO="java -Xss${JAVA_STACK_SIZE:-512k} -classpath /usr/share/java/js.jar  org.mozilla.javascript.tools.shell.Main"
 $RHINO $DIR/build/build.js baseUrl=$DIR/build load=build profile=$DIR/../src/$profile.profile.js
 exit $?
--- a/install/ui/util/uglifyjs/uglify
+++ b/install/ui/util/uglifyjs/uglify
@@ -25,7 +25,7 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]
 
 # rhino-1.7R4 doesn't have -main option to enable CommonJS support. It was
 # replaced by -require option.
-RHINO="java -Xss${JAVA_STACK_SIZE:-512k} -classpath /usr/share/java/rhino.jar  org.mozilla.javascript.tools.shell.Main"
+RHINO="java -Xss${JAVA_STACK_SIZE:-512k} -classpath /usr/share/java/js.jar  org.mozilla.javascript.tools.shell.Main"
 if [ `$RHINO --help | grep -e -require | wc -l` -gt 0 ] ; then
     $RHINO -require $DIR/uglify-js.js $@
 else
