--- a/ipaplatform/debian/paths.py
+++ b/ipaplatform/debian/paths.py
@@ -90,12 +90,12 @@ class DebianPathNamespace(BasePathNamesp
     FONT_AWESOME_DIR = "font-awesome"
     VAR_KERBEROS_KRB5KDC_DIR = "/var/lib/krb5kdc/"
     VAR_KRB5KDC_K5_REALM = "/var/lib/krb5kdc/.k5."
-    CACERT_PEM = "/var/lib/krb5kdc/cacert.pem"
+    CACERT_PEM = "/var/lib/ipa/certs/cacert.pem"
     KRB5KDC_KADM5_ACL = "/etc/krb5kdc/kadm5.acl"
     KRB5KDC_KADM5_KEYTAB = "/etc/krb5kdc/kadm5.keytab"
     KRB5KDC_KDC_CONF = "/etc/krb5kdc/kdc.conf"
-    KDC_CERT = "/var/lib/krb5kdc/kdc.crt"
-    KDC_KEY = "/var/lib/krb5kdc/kdc.key"
+    KDC_CERT = "/var/lib/ipa/certs/kdc.crt"
+    KDC_KEY = "/var/lib/ipa/certs/kdc.key"
     VAR_LOG_HTTPD_DIR = "/var/log/apache2"
     VAR_LOG_HTTPD_ERROR = "/var/log/apache2/error.log"
     NAMED_RUN = "/var/cache/bind/named.run"
