--- a/install/share/bind.named.conf.template
+++ b/install/share/bind.named.conf.template
@@ -3,7 +3,7 @@ options {
 	listen-on-v6 {any;};
 
 	// Put files that named is allowed to write in the data/ directory:
-	directory "/var/named"; // the default
+	directory "/var/cache/bind"; // the default
 	dump-file		"data/cache_dump.db";
 	statistics-file		"data/named_stats.txt";
 	memstatistics-file	"data/named_mem_stats.txt";
@@ -14,7 +14,7 @@ options {
 	// Any host is permitted to issue recursive queries
 	allow-recursion { any; };
 
-	tkey-gssapi-keytab "/etc/named.keytab";
+	tkey-gssapi-keytab "/etc/bind/named.keytab";
 	pid-file "/run/named/named.pid";
 
 	dnssec-enable yes;
@@ -37,7 +37,7 @@ zone "." IN {
 	file "named.ca";
 };
 
-include "/etc/named.rfc1912.zones";
+include "/etc/bind/named.conf.default-zones";
 
 dynamic-db "ipa" {
 	library "ldap.so";
