commit 4e9af2078fa4cdc69f97892b1055ac4a1558845c
Author: Timo Aaltonen <tjaalton@debian.org>
Date:   Tue Oct 6 16:35:24 2015 +0300

    ipaplatform: Add SECURE_NFS_VAR to constants

--- a/ipa-client/ipa-install/ipa-client-automount
+++ b/ipa-client/ipa-install/ipa-client-automount
@@ -38,6 +38,7 @@ from ipaclient import ipadiscovery
 from ipaclient import ipachangeconf
 from ipapython.ipa_log_manager import *
 from ipapython.dn import DN
+from ipaplatform.constants import constants
 from ipaplatform.tasks import tasks
 from ipaplatform import services
 from ipaplatform.paths import paths
@@ -307,7 +308,7 @@ def configure_nfs(fstore, statestore):
     Configure secure NFS
     """
     replacevars = {
-        'SECURE_NFS': 'yes',
+        constants.SECURE_NFS_VAR: 'yes',
     }
     ipautil.backup_config_and_replace_variables(fstore,
         paths.SYSCONFIG_NFS, replacevars=replacevars)
--- a/ipaplatform/base/constants.py
+++ b/ipaplatform/base/constants.py
@@ -11,3 +11,5 @@ class BaseConstantsNamespace(object):
     HTTPD_USER = "apache"
     IPA_DNS_PACKAGE_NAME = "freeipa-server-dns"
     NAMED_USER = "named"
+    # nfsd init variable used to enable kerberized NFS
+    SECURE_NFS_VAR = "SECURE_NFS"
