--- a/ipaplatform/debian/paths.py
+++ b/ipaplatform/debian/paths.py
@@ -65,6 +65,7 @@ class DebianPathNamespace(BasePathNamesp
     UPDATE_CA_TRUST = "/usr/sbin/update-ca-certificates"
     BIND_LDAP_DNS_IPA_WORKDIR = "/var/cache/bind/dyndb-ldap/ipa/"
     BIND_LDAP_DNS_ZONE_WORKDIR = "/var/cache/bind/dyndb-ldap/ipa/master/"
+    LIBARCH = "/{0}".format(MULTIARCH)
     LIBSOFTHSM2_SO = "/usr/lib/softhsm/libsofthsm2.so"
     PAM_KRB5_SO = "/usr/lib/{0}/security/pam_krb5.so".format(MULTIARCH)
     LIB_SYSTEMD_SYSTEMD_DIR = "/lib/systemd/system/"
--- a/ipaserver/install/ldapupdate.py
+++ b/ipaserver/install/ldapupdate.py
@@ -42,6 +42,7 @@ from ipalib import api, create_api
 from ipalib import constants
 from ipaplatform.paths import paths
 from ipapython.dn import DN
+from ipaplatform import NAME
 
 if six.PY3:
     unicode = str
@@ -327,12 +328,15 @@ class LDAPUpdate(object):
            etc.  Determine if a suffix is needed based on the current
            architecture.
         """
-        bits = platform.architecture()[0]
-
-        if bits == "64bit":
-            return "64"
+        if NAME is 'debian':
+            return paths.LIBARCH
         else:
-            return ""
+            bits = platform.architecture()[0]
+
+            if bits == "64bit":
+                return "64"
+            else:
+                return ""
 
     def _template_str(self, s):
         try:
