diff --git a/ipa-client/ipa-install/ipa-client-automount b/ipa-client/ipa-install/ipa-client-automount
index 3952642..e7b843e 100755
--- a/ipa-client/ipa-install/ipa-client-automount
+++ b/ipa-client/ipa-install/ipa-client-automount
@@ -39,10 +39,10 @@ from ipapython.ipa_log_manager import *
 from ipapython.dn import DN
 from ipapython import services as ipaservices
 
-AUTOFS_CONF = '/etc/sysconfig/autofs'
+AUTOFS_CONF = '/etc/default/autofs'
 NSSWITCH_CONF = '/etc/nsswitch.conf'
 AUTOFS_LDAP_AUTH = '/etc/autofs_ldap_auth.conf'
-NFS_CONF = '/etc/sysconfig/nfs'
+NFS_CONF = '/etc/default/nfs-common'
 IDMAPD_CONF = '/etc/idmapd.conf'
 
 def parse_options():
@@ -309,7 +309,7 @@ def configure_nfs(fstore, statestore):
     Configure secure NFS
     """
     replacevars = {
-        'SECURE_NFS': 'yes',
+        'NEED_GSSD': 'yes',
     }
     ipautil.backup_config_and_replace_variables(fstore,
         NFS_CONF, replacevars=replacevars)
