Author: Timo Aaltonen <tjaalton@ubuntu.com>
Date:   Mon Jan 2 16:09:40 2012 +0200

use the debian layout when installing python modules

--- a/Makefile.python.am
+++ b/Makefile.python.am
@@ -28,6 +28,7 @@ install-exec-local: $(top_builddir)/ipas
 		        --prefix "$(DESTDIR)$(prefix)" \
 		        --single-version-externally-managed \
 		        --record "$(DESTDIR)$(pkgpythondir)/install_files.txt" \
+			--install-layout=deb \
 		        --optimize 1; \
 	fi
 
