Author: Timo Aaltonen <tjaalton@ubuntu.com>
Date:   Mon Jan 2 16:09:40 2012 +0200

use the debian layout when installing python modules

--- a/Makefile
+++ b/Makefile
@@ -96,11 +96,11 @@ client-install: client client-dirs
 	done
 	cd install/po && $(MAKE) install || exit 1;
 	if [ "$(DESTDIR)" = "" ]; then \
-		$(PYTHON) setup-client.py install; \
-		(cd ipaplatform && $(PYTHON) setup.py install); \
+		$(PYTHON) setup-client.py install --install-layout=deb; \
+		(cd ipaplatform && $(PYTHON) setup.py install --install-layout=deb); \
 	else \
-		$(PYTHON) setup-client.py install --root $(DESTDIR); \
-		(cd ipaplatform && $(PYTHON) setup.py install --root $(DESTDIR)); \
+		$(PYTHON) setup-client.py install --root $(DESTDIR) --install-layout=deb; \
+		(cd ipaplatform && $(PYTHON) setup.py install --root $(DESTDIR) --install-layout=deb); \
 	fi
 
 client-dirs:
@@ -171,11 +171,11 @@ server: version-update
 
 server-install: server
 	if [ "$(DESTDIR)" = "" ]; then \
-		$(PYTHON) setup.py install; \
-		(cd ipaplatform && $(PYTHON) setup.py install); \
+		$(PYTHON) setup.py install --install-layout=deb; \
+		(cd ipaplatform && $(PYTHON) setup.py install --install-layout=deb); \
 	else \
-		$(PYTHON) setup.py install --root $(DESTDIR); \
-		(cd ipaplatform && $(PYTHON) setup.py install --root $(DESTDIR)); \
+		$(PYTHON) setup.py install --root $(DESTDIR) --install-layout=deb; \
+		(cd ipaplatform && $(PYTHON) setup.py install --root $(DESTDIR) --install-layout=deb); \
 	fi
 
 tests: version-update tests-man-autogen
@@ -186,7 +186,7 @@ tests-install: tests
 	if [ "$(DESTDIR)" = "" ]; then \
 		cd ipatests; $(PYTHON) setup.py install; \
 	else \
-		cd ipatests; $(PYTHON) setup.py install --root $(DESTDIR); \
+		cd ipatests; $(PYTHON) setup.py install --root $(DESTDIR) --install-layout=deb; \
 	fi
 	cd ipatests/man && $(MAKE) install
 
--- a/ipapython/Makefile
+++ b/ipapython/Makefile
@@ -14,7 +14,7 @@ install:
 	if [ "$(DESTDIR)" = "" ]; then \
 		python2 setup.py install; \
 	else \
-		python2 setup.py install --root $(DESTDIR); \
+		python2 setup.py install --root $(DESTDIR) --install-layout=deb; \
 	fi
 	@for subdir in $(SUBDIRS); do \
 		(cd $$subdir && $(MAKE) $@) || exit 1; \
--- a/ipapython/py_default_encoding/Makefile
+++ b/ipapython/py_default_encoding/Makefile
@@ -9,7 +9,7 @@ install:
 	if [ "$(DESTDIR)" = "" ]; then \
 		python2 setup.py install; \
 	else \
-		python2 setup.py install --root $(DESTDIR); \
+		python2 setup.py install --root $(DESTDIR) --install-layout=deb; \
 	fi
 
 clean:
