--- a/ipa-client/ipa-install/ipa-client-install
+++ b/ipa-client/ipa-install/ipa-client-install
@@ -228,7 +228,7 @@ def log_service_error(name, action, erro
     root_logger.error("%s failed to %s: %s", name, action, str(error))
 
 def nickname_exists(nickname):
-        (sout, serr, returncode) = run(["/usr/bin/certutil", "-L", "-d", "/etc/pki/nssdb", "-n", nickname], raiseonerr=False)
+        (sout, serr, returncode) = run(["/usr/bin/certutil", "-L", "-d", "sql:/etc/pki/nssdb", "-n", nickname], raiseonerr=False)
 
         if returncode == 0:
             return True
@@ -487,7 +487,7 @@ def uninstall(options, env):
     # Remove our host cert and CA cert
     if nickname_exists("IPA CA"):
         try:
-            run(["/usr/bin/certutil", "-D", "-d", "/etc/pki/nssdb", "-n", "IPA CA"])
+            run(["/usr/bin/certutil", "-D", "-d", "sql:/etc/pki/nssdb", "-n", "IPA CA"])
         except Exception, e:
             root_logger.error(
                 "Failed to remove IPA CA from /etc/pki/nssdb: %s", str(e))
@@ -515,7 +515,7 @@ def uninstall(options, env):
 
     if nickname_exists(client_nss_nickname):
         try:
-            run(["/usr/bin/certutil", "-D", "-d", "/etc/pki/nssdb", "-n", client_nss_nickname])
+            run(["/usr/bin/certutil", "-D", "-d", "sql:/etc/pki/nssdb", "-n", client_nss_nickname])
         except Exception, e:
             root_logger.error("Failed to remove %s from /etc/pki/nssdb: %s",
                 client_nss_nickname, str(e))
