--- a/ipa-client/ipa-install/ipa-client-install
+++ b/ipa-client/ipa-install/ipa-client-install
@@ -201,7 +201,7 @@ def log_service_error(name, action, erro
     root_logger.error("%s failed to %s: %s", name, action, str(error))
 
 def nickname_exists(nickname):
-        (sout, serr, returncode) = run(["/usr/bin/certutil", "-L", "-d", "/etc/pki/nssdb", "-n", nickname], raiseonerr=False)
+        (sout, serr, returncode) = run(["/usr/bin/certutil", "-L", "-d", "sql:/etc/pki/nssdb", "-n", nickname], raiseonerr=False)
 
         if returncode == 0:
             return True
@@ -365,7 +365,7 @@ def uninstall(options, env):
     # Remove our host cert and CA cert
     if nickname_exists("IPA CA"):
         try:
-            run(["/usr/bin/certutil", "-D", "-d", "/etc/pki/nssdb", "-n", "IPA CA"])
+            run(["/usr/bin/certutil", "-D", "-d", "sql:/etc/pki/nssdb", "-n", "IPA CA"])
         except Exception, e:
             root_logger.error(
                 "Failed to remove IPA CA from /etc/pki/nssdb: %s", str(e))
@@ -393,7 +393,7 @@ def uninstall(options, env):
 
     if nickname_exists(client_nss_nickname):
         try:
-            run(["/usr/bin/certutil", "-D", "-d", "/etc/pki/nssdb", "-n", client_nss_nickname])
+            run(["/usr/bin/certutil", "-D", "-d", "sql:/etc/pki/nssdb", "-n", client_nss_nickname])
         except Exception, e:
             root_logger.error("Failed to remove %s from /etc/pki/nssdb: %s",
                 client_nss_nickname, str(e))
@@ -2297,7 +2297,7 @@ def install(options, env, fstore, states
 
     # Add the CA to the default NSS database and trust it
     try:
-        run(["/usr/bin/certutil", "-A", "-d", "/etc/pki/nssdb", "-n", "IPA CA", "-t", "CT,C,C", "-a", "-i", CACERT])
+        run(["/usr/bin/certutil", "-A", "-d", "sql:/etc/pki/nssdb", "-n", "IPA CA", "-t", "CT,C,C", "-a", "-i", CACERT])
     except CalledProcessError, e:
         root_logger.info("Failed to add CA to the default NSS database.")
         return CLIENT_INSTALL_ERROR
