--- a/ipaserver/install/dogtaginstance.py
+++ b/ipaserver/install/dogtaginstance.py
@@ -34,6 +34,7 @@ from ipalib import errors
 
 from ipaplatform import services
 from ipaplatform.paths import paths
+from ipaplatform.constants import constants
 from ipapython import certmonger
 from ipapython import ipaldap
 from ipapython import ipautil
@@ -45,7 +46,7 @@ from ipaserver.install.installutils impo
 from ipapython.ipa_log_manager import log_mgr
 
 PKI_USER = "pkiuser"
-
+HTTPD_USER = constants.HTTPD_USER
 
 def get_security_domain():
     """
@@ -87,7 +88,7 @@ def export_kra_agent_pem():
             "--client-cert", filename]
     ipautil.run(args)
 
-    pent = pwd.getpwnam("apache")
+    pent = pwd.getpwnam(HTTPD_USER)
     os.chown(filename, 0, pent.pw_gid)
     os.chmod(filename, 0o440)
 
